server.py

```
@@ -553,56 +542,49 @@ async def spinner_task_func():
+                    "Code", "PromptCache", "McpEverhour", "McpAsana", "McpGmail",
+                    "Mcp", "gpt-4o"
                 ]
-                if any(keyword in (text or "").lower() for keyword in image_generation_keywords):
-                    return "ImageGen"
-                if image_urls and not any(keyword in (text or "").lower() for keyword in image_generation_keywords):
-                    return "Vision"
-                return "gpt-4o"
+                return "+".join([t for t in order if t in detected])
 
-            tag = get_response_tag()
+            tag = derive_tag([], [], user_message=text)
```

Pylint (3.3.7)

[error] Unexpected keyword argument 'user_message' in function call

(E1123)